Updated JavaScript kernel with typescript, top-level async/await, es module import, promise resolution

mishushakov opened the pull request:
- replaces ijavascript kernel with a fork that includes a compiler step and promise resolution
- removes typescript compiler step as this is now built-in in the kernel

dobrac commented:
> (node:322) ExperimentalWarning: vm.USE_MAIN_CONTEXT_DEFAULT_LOADER is an experimental feature and might change at any time\n(Use `node --trace-warnings ...` to show where the warning was created)\n

I would hide this from the user as its more related to our implementation than to the usage.

mishushakov commented:
I have changed the way ESM imports are handled, have to verify that it no longer outputs any warnings, otherwise I can add a flag to the jupyter kernel that it should run node with --no-warnings flag

0div reviewed (approved):
Looks promising! 

There's one thing i do not understand in the ijavascript fork: how do the [dependencies change ](https://github.com/n-riesco/ijavascript/compare/master...e2b-dev:ijavascript:master#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519) affect the behaviour in it? I don't see them being imported in the code changes? 

Also it would be nice to make a PR for each original repo? nice contrib and that way if it's accepted and merged we can use them again in our deps? 

I take it the code-interpeter js tests passed with this?

mishushakov commented:
Hey Jonas,
I have replaced the said dependencies with our own versions with some patches. Main difference is we now have a pre-compile step which transforms the source code and the kernel now awaits asynchronous code.

PR to the original repo would be not very productive as the original projects were abandoned, which is why we had to fork them in the first place.

Existing tests pass, I removed the negative scenarios that should no longer be failing. New tests will be in a separate PR. I also confirmed there are indeed no warnings when trying to import the modules. Currently waiting for a deploy of the updated template-manager so we can update the code interpreter template.
